snaps/reference/permissions.md

```diff
@@ -24,12 +24,30 @@ manifest file:
 
 ## Endowments
 
+### Snap-defined timeouts
+
+Many of the endowments defined below entail having MetaMask run arbitrary code defined in the Snap. The default execution timeout is 60000ms, that is one minute.
+
+This execution timeout can be modified by adding a caveat `maxRequestTime` to the permission. It can take values from `5000` (5 seconds) all the way to `180000` (3 minutes). The following endowments accept this caveat:
+
+- `endowment:cronjob`
+- `endowment:keyring`
+- `endowment:lifecycle-hooks`
+- `endowment:name-lookup`
+- `endowment:page-home`
+- `endowment:rpc`
+- `endowment:transaction-insight`
+
 ### `endowment:cronjob`
 
 To run periodic actions for the user (cron jobs), a Snap must request the `endowment:cronjob` permission.
 This permission allows the Snap to specify cron jobs that trigger the
 [`onCronjob`](../reference/entry-points.md#oncronjob) entry point.
 
+:::tip
+You can increase the cron job's execution time by using [Snap-defined timeouts](#snap-defined-timeouts).
```

Comment:
I think every instance of "increase" should be "modify" because you can both increase or decrease it

Also I would say "execution limit" not "time."